test/e2e/e2ethanos/services.go

```diff
@@ -91,6 +91,28 @@ func defaultPromHttpConfig() string {
 `
 }
 
+const nginxImage = "docker.io/nginx:1.21.1-alpine"
+
+func NewStaticMetricsServer(e e2e.Environment, name string, metrics []byte) *e2emon.InstrumentedRunnable {
```

Comment:
Maybe it's a good candidate for the e2e repo itself? I was thinking about implementing this same thing myself today :smile:

Reply:
I'm happy to add it there and update the code here after it gets merged, like with the `Between` matcher, to avoid blocking progress. 👍 

Will open a PR over there and link it here.

Reply:
Opened the PR upstream: https://github.com/efficientgo/e2e/pull/67

Reply:
@GiedriusS as the upstream PR was merged, I updated the code here to use it in https://github.com/thanos-io/thanos/pull/6377/commits/1c34a99804ff840bb9c13f4fd71c1f28e15d7905.